examples/docker_aws_lambda_example/producer_example.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"context"
+	"fmt"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"os"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+var p *kafka.Producer
+
+func main() {
+	// Reference: https://github.com/aws/aws-lambda-go/issues/318#issuecomment-1019318919
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		fmt.Printf("waiting for the SIGTERM\n")
+		s := <-sigs
+		fmt.Printf("received signal %s\n", s)
+		if p != nil {
+			p.Close()
+		}
+		fmt.Printf("done\n")
+	}()
+
+	lambda.Start(HandleRequest)
+}
+
+// HandleRequest handles creating producer and
+// producing messages.
+func HandleRequest() error {
+	broker := os.Getenv("BOOTSTRAP_SERVERS")
+	topic := os.Getenv("TOPIC")
+	ccloudAPIKey := os.Getenv("CCLOUDAPIKEY")
+	ccloudAPISecret := os.Getenv("CCLOUDAPISECRET")
+
+	var err error
+
+	if p == nil {
+		p, err = kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers": broker,
+			"sasl.mechanisms":   "PLAIN",
+			"security.protocol": "SASL_SSL",
+			"sasl.username":     ccloudAPIKey,
+			"sasl.password":     ccloudAPISecret,
+		})
+
+		if err != nil {
+			fmt.Printf("Failed to create producer: %s\n", err)
+			os.Exit(1)
+		}
+
+		fmt.Printf("Created Producer %v\n", p)
+	}
+
+	a, err := kafka.NewAdminClientFromProducer(p)
+	defer a.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	maxDur, err := time.ParseDuration("60s")
+	if err != nil {
+		panic("ParseDuration(60s)")
+	}
+
+	// Create topics if it doesn't exist
+	results, err := a.CreateTopics(
+		ctx,
+		[]kafka.TopicSpecification{{
+			Topic:         topic,
+			NumPartitions: 1}},
+		kafka.SetAdminOperationTimeout(maxDur))
+	if err != nil {
+		fmt.Printf("Failed to create topic: %v\n", err)
+		return err
```

Comment:
you don't want to quit if there is a "topic already exists" error, otherwise this lambda function can only be used once.

Reply:
need some sort of check like:

https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/examples/ExactlyOnce/Program.cs#L233

Reply:
(if there is an error that isn't topic already exists, then print the error and exit)

Reply:
ahh, ok, we did .NET differently.

maybe change this to `CreateTopics request failed: `